homeassistant/components/notify/lametric.py

```diff
@@ -0,0 +1,96 @@
+"""
+Notifier for LaMetric time.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.lametric/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_TARGET, ATTR_DATA, PLATFORM_SCHEMA, BaseNotificationService)
+from homeassistant.const import CONF_ICON
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.lametric import DOMAIN
+
+REQUIREMENTS = ['lmnotify==0.0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DISPLAY_TIME = "display_time"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ICON, default="i555"): cv.string,
+    vol.Optional(CONF_DISPLAY_TIME, default=10): cv.positive_int,
+})
+
+
+# pylint: disable=unused-variable
+def get_service(hass, config, discovery_info=None):
+    """Get the Slack notification service."""
+    hlmn = hass.data.get(DOMAIN)
+    try:
+        return LaMetricNotificationService(hlmn,
+                                           config[CONF_ICON],
+                                           config[CONF_DISPLAY_TIME] * 1000)
+    except HomeAssistantError:
+        _LOGGER.exception("Could not configure LaMetric notifier")
+        return None
+
+
+class LaMetricNotificationService(BaseNotificationService):
+    """Implement the notification service for LaMetric."""
+
+    def __init__(self, hasslametricmanager, icon, display_time):
+        """Initialize the service."""
+        self.hasslametricmanager = hasslametricmanager
+        self._icon = icon
+        self._display_time = display_time
+
+    # pylint: disable=broad-except
+    def send_message(self, message="", **kwargs):
+        """Send a message to some LaMetric deviced."""
+        from lmnotify import SimpleFrame, Sound, Model
+
+        targets = kwargs.get(ATTR_TARGET)
+        data = kwargs.get(ATTR_DATA)
+        _LOGGER.debug("Targets/Data: %s/%s", targets, data)
+        icon = self._icon
+        sound = None
+
+        # User-defined icon?
+        if data is not None:
+            if "icon" in data:
+                icon = data["icon"]
+            if "sound" in data:
+                try:
+                    sound = Sound(category="notifications",
+                                  sound_id=data["sound"])
+                    _LOGGER.debug("Adding notification sound %s",
+                                  data["sound"])
+                except Exception:
```

Comment:
The possible exceptions aren't documented. I've already seen oauth exceptions, but there can be all kind of other exceptions. 
I can remove the except completely. Personally I don't think this is a better solution.